### ISSUE TITLE
add verified public key types

### ISSUE THREAD

**Issue by stevedw** [briansmith/ring#1601]
Is it possible to add parsed public key types that can be stored and used repeatedly, rather than having to use the unparsedpublickey  or similar?

It means having to parse and verify the bytes on each verify call - which seems an avoidable overhead.

**Comment by briansmith** [comment]
Yes, it is very possible but I don't expect it to bring much efficiency. I started to do it for the public key component of keypairs, e.g. `rsa::PublicKey`, but there's no API yet for parsing only a public key in some format and getting a `KeyPair`.
